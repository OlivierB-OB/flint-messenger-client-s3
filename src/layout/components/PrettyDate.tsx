import React, { Fragment } from 'react';

export interface IPrettyDateProps {
  date: string;
}

const fdate = new Intl.DateTimeFormat('en-US', {
  weekday: 'long',
  year: 'numeric',
  month: 'long',
  day: 'numeric',
});

const ftime = new Intl.DateTimeFormat('en-US', {
  hour: 'numeric',
  minute: 'numeric',
  hour12: false,
});

export function PrettyDate({ date }: IPrettyDateProps) {
  const dateObject = new Date(date);
  return (
    <Fragment>
      {fdate.format(dateObject)}
      {' at '}
      {ftime.format(dateObject)}
    </Fragment>
  );
}
