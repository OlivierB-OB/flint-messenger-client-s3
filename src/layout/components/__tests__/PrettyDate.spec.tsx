import React from 'react';
import { shallow } from 'enzyme';
import { PrettyDate } from '../PrettyDate';

describe('PrettyDate', () => {
  it('should display the date', async () => {
    const component = shallow(<PrettyDate date="2020-06-07T11:53:32.000Z" />);
    expect(component).toMatchSnapshot();
  });
});
