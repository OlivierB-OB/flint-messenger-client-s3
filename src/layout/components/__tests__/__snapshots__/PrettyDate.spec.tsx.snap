// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrettyDate should display the date 1`] = `
<Fragment>
  Sunday, June 7, 2020
   at 
  13:53
</Fragment>
`;
